src/scripts/mint.ts

```diff
@@ -0,0 +1,305 @@
+      } else {
+        allowanceButton.innerText = allowanceButton.innerText.replace(appendableText, "");
+      }
+    }, 500);
+  })();
+
+  void (() => {
+    publicClient.watchBlocks({
+      onBlock: async () => {
+        try {
+          const collateralAddress = collateralRecord[selectedCollateralIndex];
+          const web3Client = getConnectedClient();
+
+          if (allowanceButton !== null)
+            allowanceButton.disabled = web3Client === null || !collateralAddress || !web3Client.account || isButtonInteractionsDisabled;
+          if (mintButton !== null) mintButton.disabled = web3Client === null || !collateralAddress || !web3Client.account || isButtonInteractionsDisabled;
+
+          if (collateralAddress && web3Client && web3Client.account) {
+            await check(collateralAddress, web3Client);
+          }
+        } catch (error) {
+          const err = error as Error;
+          toastActions.showToast({
+            toastType: "error",
+            msg: err.message,
+          });
+        }
+      },
+    });
+  })();
+}
+
+async function check(collateralAddress: `0x${string}`, web3Client: ReturnType<typeof getConnectedClient>) {
+  const collateralDecimals = await getTokenDecimals(collateralAddress);
+  const governanceDecimals = await getTokenDecimals(ubqAddress);
+  const maxCollatIn = parseUnits(maxCollateralIn.toString(), collateralDecimals);
+  const maxGovernIn = parseUnits(maxGovernanceIn.toString(), governanceDecimals);
+  const allowance0 = web3Client?.account ? await getAllowance(collateralAddress, web3Client.account.address, diamondAddress) : BigInt(0);
+  const allowance1 = web3Client?.account ? await getAllowance(ubqAddress, web3Client.account.address, diamondAddress) : BigInt(0);
+  const isAllowed = allowance0 >= maxCollatIn && (!isOneToOne ? allowance1 >= maxGovernIn : true);
+  updateUiBasedOnAllowance(isAllowed);
+}
+
+function updateUiBasedOnAllowance(isAllowed: boolean) {
+  if (isAllowed) {
+    if (mintButton !== null && mintButton.classList.contains("hidden")) {
+      mintButton.classList.remove("hidden");
+      mintButton.classList.add("flex");
+    }
+    if (allowanceButton !== null && !allowanceButton.classList.contains("hidden")) {
+      allowanceButton.classList.add("hidden");
+      allowanceButton.classList.remove("flex");
+    }
+  } else {
+    if (mintButton !== null && !mintButton.classList.contains("hidden")) {
+      mintButton.classList.add("hidden");
+      mintButton.classList.remove("flex");
+    }
+    if (allowanceButton !== null && allowanceButton.classList.contains("hidden")) {
+      allowanceButton.classList.remove("hidden");
+      allowanceButton.classList.add("flex");
+    }
+  }
+}
+
+export async function initCollateralList() {
+  if (collateralSelect !== null && window.location.pathname.includes("mint")) {
+    const collaterals = await getAllCollaterals();
```

Comment:
A good idea is to share the collaterals data with the entire app, this way we reduce our RPC calls, avoid limit rates and make the app faster when switching between pages.